openmls/src/group/core_group/new_from_external_init.rs

```diff
@@ -84,20 +85,15 @@ impl CoreGroup {
         let params_credential_with_key = params
             .take_credential_with_key()
             .ok_or(ExternalCommitError::MissingCredential)?;
-        for Member {
-            index,
-            signature_key,
-            ..
-        } in public_group.members()
-        {
-            if signature_key == params_credential_with_key.signature_key.as_slice() {
-                let remove_proposal = Proposal::Remove(RemoveProposal { removed: index });
-                inline_proposals.push(remove_proposal);
-                break;
-            };
-        }
+        if let Some(us) = public_group.members().find(|member| {
```

Comment:
This is only a refactoring. While debugging, I suspected that the leaf node index is not calculated correctly because we always used "free_leaf_index" even for "us". However, there are two `free_leaf_index` methods and the one used here is subtly different. Thus, I renamed it.